plugins/lua/ts_lua_client_request.c

```diff
@@ -399,6 +403,65 @@ ts_lua_client_request_get_headers(lua_State *L)
   return 1;
 }
 
+static int
+ts_lua_client_request_get_header_block(lua_State *L)
+{
+  TSIOBuffer       output_buffer;
+  TSIOBufferReader reader;
+  int              total_avail;
+
+  TSIOBufferBlock block;
+  const char     *block_start;
+  int64_t         block_avail;
+
+  char   *output_string;
+  int64_t output_len;
+
+  ts_lua_http_ctx *http_ctx;
+
+  GET_HTTP_CONTEXT(http_ctx, L);
+
+  output_buffer = TSIOBufferCreate();
+  reader        = TSIOBufferReaderAlloc(output_buffer);
+
+  TSMimeHdrPrint(http_ctx->client_request_hdrp, output_buffer);
+
+  total_avail = TSIOBufferReaderAvail(reader);
+
+  output_string = static_cast<char *>(TSmalloc(total_avail + 1));
```

Comment:
we also moved from c to cc
so some syntax is to be different 

 output_string = (char *)TSmalloc(total_avail + 1);